actors/migration/reward.go

```diff
@@ -3,35 +3,153 @@ package migration
 import (
 	"context"
 
+	"golang.org/x/xerrors"
+
 	"github.com/filecoin-project/go-state-types/abi"
 	"github.com/filecoin-project/go-state-types/big"
 	reward0 "github.com/filecoin-project/specs-actors/actors/builtin/reward"
 	cid "github.com/ipfs/go-cid"
 	cbor "github.com/ipfs/go-ipld-cbor"
 
+	builtin2 "github.com/filecoin-project/specs-actors/v2/actors/builtin"
+	power2 "github.com/filecoin-project/specs-actors/v2/actors/builtin/power"
 	reward2 "github.com/filecoin-project/specs-actors/v2/actors/builtin/reward"
+	"github.com/filecoin-project/specs-actors/v2/actors/states"
+	math2 "github.com/filecoin-project/specs-actors/v2/actors/util/math"
 	smoothing2 "github.com/filecoin-project/specs-actors/v2/actors/util/smoothing"
 )
 
+var (
+	// Q.128 coefficients used to compute new effective network time
+	// (g*Tau is the BaselineExponent, B0 is the BaselineInitialValue)
+	gTauOverB0  big.Int
+	oneOverGTau big.Int
+)
+
+func init() {
+	constStrs := []string{
+		"77669179383316",
+		"16058975841646949912583456815941211282956743",
+	}
+	constBigs := math2.Parse(constStrs)
+	gTauOverB0 = big.NewFromGo(constBigs[0]) // Q.128
+	oneOverGTau = big.NewFromGo(constBigs[1])
+}
+
 type rewardMigrator struct {
+	actorsOut *states.Tree
 }
 
-func (m *rewardMigrator) MigrateState(ctx context.Context, store cbor.IpldStore, head cid.Cid, _ MigrationInfo) (cid.Cid, abi.TokenAmount, error) {
+func (m *rewardMigrator) MigrateState(ctx context.Context, store cbor.IpldStore, head cid.Cid, migInfo MigrationInfo) (cid.Cid, abi.TokenAmount, error) {
 	var inState reward0.State
 	if err := store.Get(ctx, head, &inState); err != nil {
 		return cid.Undef, big.Zero(), err
 	}
 
+	// Get migrated power actor's ThisEpochRawBytePower
+	outPowerAct, found, err := m.actorsOut.GetActor(builtin2.StoragePowerActorAddr)
+	if err != nil {
+		return cid.Undef, big.Zero(), err
+	}
+	if !found {
+		return cid.Undef, big.Zero(), xerrors.Errorf("could not find power actor in migrated state")
+	}
+	var outPowerSt power2.State
+	if err := store.Get(ctx, outPowerAct.Head, &outPowerSt); err != nil {
+		return cid.Undef, big.Zero(), xerrors.Errorf("failed to read migrated power actor state %w", err)
+	}
+	thisEpochRawBytePower := outPowerSt.ThisEpochRawBytePower
+
+	one := big.Lsh(big.NewInt(1), math2.Precision128) // Q.128
+	// Set baseline to the value it would be at the end of the prior epoch if
+	// current parameters had been in place at network start
+	outThisEpochBaselinePower := math2.ExpBySquaring(reward2.BaselineExponent, int64(migInfo.priorEpoch)) // Q.128
+	outThisEpochBaselinePower = big.Mul(reward2.BaselineInitialValue, outThisEpochBaselinePower)          // Q.0 * Q.128 => Q.128
```

Comment:
Is it safe to assume that the `reward2.Baseline...` constants are the new ones (e.g. BaselineInitialValue is 2 888 888 880 000 000 000)?

Reply:
Yes they are the new ones and that is the value of BaselineInitialValue.  For completeness
`reward2.BaselineExponent = big.MustFromString("340282591298641078465964189926313473653") // Q.128`
and `reward2.Lambda` and `reward2.ExpLamSubOne` have not changed from `reward0` values